syft/pkg/cataloger/package_url_test.go

```diff
@@ -136,3 +130,10 @@ func TestPackageURL(t *testing.T) {
 		})
 	}
 }
+
+func intRef(i ...int) *int {
```

Comment:
This is clever. But I think it would be more straightforward to take only a single `int` and return a pointer to it, and in cases where the consumer just needs `nil`, to use `nil` directly, since seeing `Epoch: intRef()` is less immediately obvious than seeing `Epoch: nil`.

Reply:
updated in https://github.com/anchore/syft/pull/427/commits/afbd8f8ea02746624e037801d288da66df1a1314

Reply:
Awesome, looks good!